src/backend/lang_c/c_simple_ast.rs

```diff
@@ -0,0 +1,209 @@
+use std::{default, iter, fmt};
+use std::collections::LinkedList;
+
+use super::c_simple;
+use super::c_simple::{CAST, CASTNode};
+use frontend::radeco_containers::RadecoFunction;
+use petgraph::graph::{Graph, NodeIndex, EdgeIndex, Edges, EdgeReference};
+use petgraph::visit::EdgeRef;
+use petgraph::{EdgeDirection, Direction, Directed};
+
+#[derive(Debug, Clone, PartialEq)]
+enum SimpleCASTNode {
+    Action(ActionNode),
+    Value(ValueNode),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum ActionNode {
+    Assignment,
+    Call,
+    Return,
+    If,
+    Goto,
+}
+
+// TODO
+#[derive(Debug, Clone, PartialEq)]
+enum ValueNode {
+    /// The string is the name of variable
+    Variable(String),
+    Constant,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum SimpleCASTEdge {
+    Action(ActionEdge),
+    Value(ValueEdge),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum ActionEdge {
+    IfThen,
+    IfElse,
+    FuncCall,
+    FuncRet,
+    Normal,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum ValueEdge {
+    AssignSrc,
+    AssignDst,
+    Arg(u8),
+    GotoDst,
+}
+
+struct SimpleCAST {
+    /// Name of function of this AST
+    fname: String,
+    entry: NodeIndex,
+    ast: Graph<SimpleCASTNode, SimpleCASTEdge>,
+}
+
+
+impl SimpleCASTNode {
+    fn is_exit(&self) -> bool {
+        match self {
+            &SimpleCASTNode::Action(ActionNode::Return) => true,
+            _ => false,
+        }
+    }
+}
+
+/// Returns nodes which is connected with given type of edge
+fn neighbors_by_edge(edges: &Vec<EdgeReference<SimpleCASTEdge>>, ty: &SimpleCASTEdge) -> Vec<NodeIndex> {
```

Comment:
Hey @kriw , any specific reason why this method is not part of the `impl` for SimpleCAST?